frontend/packages/dev-console/src/components/pipelines/list-page/PipelineRow.tsx

```diff
@@ -72,4 +80,8 @@ const PipelineRow: RowFunction<Pipeline> = ({ obj, index, key, style }) => {
   );
 };
 
+const PipelineRow: RowFunction<Pipeline> = ({ obj, index, style }) => {
+  return <PipelineTableRow obj={obj} index={index} style={style} />;
+};
+
```

Comment:
This is my biggest concern of this PR. This is not the pattern we are used to for Table rows. Perhaps we need Christian to comment on the architecture of this decision. I'm going to investigate an alternative for the table rows.

Reply:
I wonder if we just override my solution for the hook injection of the user label. I wrapped the Kebab component instead of the whole row. Most of the content is unaffected by this state and thus doesn't need this change.

Recommend you try that, maybe another wrapper 🤔 Maybe a prop 🤔 Consider this path instead of making the whole row a component. Let me know if you want to discuss options.

Reply:
Ah - crud. This may be actually more difficult. This is done all up front AND on scroll (if there are enough to be virtually rendered). This solution may need to focus on a hover of the Kebab. Perhaps we can use a `useEffect` and a `useRef` hook on a wrapping div 🤔

Reply:
As ```PipelineRow``` is not a ```FC``` so, I am not able to use hooks. That is why I make this change.

Reply:
I understand that @vikram-raj This pattern of creating a component off a function to use hooks... especially for the row component and always on the row component is a performance issue. We need to do better otherwise we'll just be trading bugs.

Reply:
@andrewballantyne Updated this. Now, I wrap only the pipeline kebab component instead of wrapping the whole row component.